### PR TITLE
feat: Contacts verification and merging

### PR DESCRIPTION
**Pull request by tapiarafael**
## Proposed changes (including videos or screenshots)

 - Adds EE capability to verify and merge similar contacts. Verified contacts are considered "similar" when their channels' fields and values match.

## Further comments

[SCI-41](https://rocketchat.atlassian.net/browse/SCI-41)

[SCI-41]: https://rocketchat.atlassian.net/browse/SCI-41?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/server/services/omnichannel/service.ts [hunk 1]
```diff
@@ -60,4 +62,32 @@ export class OmnichannelService extends ServiceClassInternal implements IOmnicha
 		const currentMonth = moment.utc().format('YYYY-MM');
 		return room.v?.activity?.includes(currentMonth) || !(await License.shouldPreventAction('monthlyActiveContacts'));
 	}
+
+	async mergeContacts(contactId: string, channel: ILivechatContactChannel): Promise<ILivechatContact | null> {
```
Comment (matheusbsilva137) [inline apps/meteor/server/services/omnichannel/service.ts:66]: Should we move this to under `overwriteClassOnLicense` too? wdyt @pierre-lehnen-rc ?

Comment (tapiarafael) [reply apps/meteor/server/services/omnichannel/service.ts:66]: I believe that would be better, since the merging feature is enterprise only

Comment (matheusbsilva137) [reply apps/meteor/server/services/omnichannel/service.ts]: Great, thanks! I moved it to EE too

File: apps/meteor/tests/unit/app/importer/server/userConverter.spec.ts [hunk 2]
```diff
@@ -120,12 +121,14 @@ describe('User Converter', () => {
...
-		sinon.stub(converter, 'hashPassword');
-		converter.generateTempPassword.returns('tempPassword');
-		converter.hashPassword.callsFake((pass: string) => `hashed=${pass}`);
+
+		before(() => {
+			sinon.stub(converter, 'generateTempPassword');
+			sinon.stub(converter, 'hashPassword');
+			converter.generateTempPassword.returns('tempPassword');
+			converter.hashPassword.callsFake((pass: string) => `hashed=${pass}`);
+		});
```
Comment (MarcosSpessatto) [inline apps/meteor/tests/unit/app/importer/server/userConverter.spec.ts:131]: Is this related to the PR? 🤔 If not, please remove it

File: packages/core-typings/src/IRoom.ts [hunk 3]
```diff
@@ -320,6 +320,9 @@ export interface IOmnichannelRoom extends IOmnichannelGenericRoom {
 	// which is controlled by Livechat_auto_transfer_chat_timeout setting
 	autoTransferredAt?: Date;
 	autoTransferOngoing?: boolean;
+
+	// Whether the contact is verified or not
```
Comment (MarcosSpessatto) [inline packages/core-typings/src/IRoom.ts:324]: Do we really need this comment? 🤔

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

File: apps/meteor/tests/unit/app/importer/server/userConverter.spec.ts (hunk 2 above)
Comment (matheusbsilva137) [reply apps/meteor/tests/unit/app/importer/server/userConverter.spec.ts:131]: Unit tests were failing here, so I had to add this to make sure the new unit tests were ok in this PR
@pierre-lehnen-rc will apply this fix in another PR (while this fix isn't merged, unit tests will fail) , so I'll remove this from here

**Review by pierre-lehnen-rc — APPROVED** [review]
